### PR TITLE
1390 - Adapt forward pass of new batch object

### LINKED ISSUE #1390: Adapt model and TargetAuxCalc forward to new batch item
### Describe the task. It can be a feature, documentation, etc.

After the stream data refactoring, the new batch object has to be consumed by the model and the target-aux-calculator. 
A new function is needed to transfer the batch to device and downstream forward functions have to be adapted.

### Hedgedoc URL, if you are keeping notes, plots, logs in hedgedoc.

_No response_

### Area

- [ ] datasets, data readers, data preparation and transfer
- [x] model
- [ ] science
- [ ] infrastructure and engineering
- [ ] evaluation, export and visualization
- [ ] documentation

### PR DESCRIPTION
**Pull request by Jubeku**
## Description

After the stream data refactoring, the new batch object has to be consumed by the model and the target-aux-calculator.
A new function is needed to transfer the batch to device and downstream forward functions have to be adapted.

TODO adapt validate and inference
TODO test forecasting and multiple stream because predict changed substantially

## Issue Number
 
Closes #1390 

Is this PR a draft? Mark it as draft.

## Checklist before asking for review

-   [ ] I have performed a self-review of my code
-   [ ] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [ ] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [ ] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

File: src/weathergen/train/trainer.py [hunk 1]
```diff
@@ -535,40 +535,23 @@ def train(self, mini_epoch):
...
-                    source_cell_lens = [b.to(self.device) for b in source_cell_lens]
-                    target_coords_idxs = view.target_coords_idx
-                    target_coords_idxs = [
-                        [b.to(self.device) for b in bf] for bf in target_coords_idxs
-                    ]
                     outputs.append(
                         self.model(
                             self.model_params,
-                            (streams_data, source_cell_lens, target_coords_idxs),
+                            (view.streams_data, view.source_cell_lens, view.target_coords_idx),
```
Comment (Jubeku) [inline src/weathergen/train/trainer.py]: Should we pass `view` directly and adapt the model forward [here](https://github.com/ecmwf/WeatherGenerator/pull/1391/files#diff-32337781af2a469f8a1131c4327330f999a45a9032bbed5c2a12a69b20f6ffefL582) accordingly?

File: src/weathergen/model/model.py [hunk 2]
```diff
@@ -865,10 +860,10 @@ def predict(
             assert isinstance(tte_kv, torch.nn.Identity)
 
             # lens for varlen attention
-            tcs_lens = target_coords_idxs[idx][fstep]
+            tcs_lens = target_coords_idxs[fstep]
```
Comment (Jubeku) [inline src/weathergen/model/model.py]: It looks like we miss here that `target_coords_idxs` must be a function of the stream.

File: src/weathergen/train/trainer.py (hunk 1 above)
Comment (clessig) [reply src/weathergen/train/trainer.py]: view is student/teacher specific. It should be called sample except for code that is student/teacher specific (which is only in some parts of the loss and some specific masking code).

Comment (Jubeku) [reply src/weathergen/train/trainer.py]: I am still wondering whether to pass `sample` instead of `(sample.streams_data, sample.source_cell_lens, sample.target_coords_idx)`?
